[import-w3c-tests] check all baselines when deleting dangling baselines

gsnedders opened the pull request:
#### 69c47f823b0e278ba4a79e9661133153e6b334e5
<pre>
[import-w3c-tests] check all baselines when deleting dangling baselines
<a href="https://bugs.webkit.org/show_bug.cgi?id=264735">https://bugs.webkit.org/show_bug.cgi?id=264735</a>

Reviewed by Jonathan Bedard.

Previously, we found the first baseline for the current port, and
preserved that baseline. This is, of course, wrong: we need to preserve
baselines for all platforms, including (and especially) the generic
baseline.

* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.remove_dangling_expectations):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
(TestImporterTest.import_downloaded_tests):

Canonical link: <a href="https://commits.webkit.org/270899@main">https://commits.webkit.org/270899@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eeafe5e5ffcb22823295afe5d683c92f7a36b03f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26718 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5332 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27958 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28930 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24429 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27163 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7159 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2729 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24338 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26980 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4149 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22939 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3646 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/26878 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3687 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23931 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29415 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24362 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24331 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29951 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3725 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1920 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27852 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5174 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4194 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3455 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4078 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/webkitpy/common/net/bugzilla/test_expectation_updater.py:
```diff
@@ -79,15 +79,15 @@ def argument_parser():
 class TestExpectationUpdater(object):
 
-    def __init__(self, host, bugzilla_id, is_bug_id=True, bot_filter_name=None, attachment_fetcher=None):
+    def __init__(self, port, bugzilla_id, is_bug_id=True, bot_filter_name=None, attachment_fetcher=None):
```
This looks like this relies off of #20551?

gsnedders replied:
See the comment above. (Yes, it does!)

gsnedders replied:
And now rebased, now that's landed.

gsnedders committed:
[import-w3c-tests] check all baselines when deleting dangling baselines
https://bugs.webkit.org/show_bug.cgi?id=264735

Reviewed by Jonathan Bedard.

Previously, we found the first baseline for the current port, and
preserved that baseline. This is, of course, wrong: we need to preserve
baselines for all platforms, including (and especially) the generic
baseline.

* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.remove_dangling_expectations):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
(TestImporterTest.import_downloaded_tests):

Canonical link: https://commits.webkit.org/270899@main
